### PR TITLE
Precompute and cache the bool value indicating if the project support…

### LINKED ISSUE #618: Large performance regression to load Picasso in AbstractSpecialFolderProjectTreePropertiesProvider
We saw a large perf regression in the CPS perf test to load Piscasso into CPS.  The AstractSpecialFolderProjectTreePropertiesProvider.CalculatePropertyValues contributes 1.4s to the CPU time during the solution load.

This is to track this part of the regression (as a part of VS bug 277029)
Trace can be found:
\\[Internal file share]\Public\cps-perf\Dev15\Picasso-cps\25810.00(wpt)\CpsPicassoEndToEnd-10-12-2016-063623\DataBackup-2016-10-12-18-36

Name                                                                                                                     	Inc %	     Inc	Inc Ct	Exc %	Exc
 microsoft.visualstudio.projectsystem.implementation!PhysicalProjectTreeProvider.ApplyPropertiesProviderExtensions       	 10.2	   1,431	 1,431	  0.0	  6
+ microsoft.visualstudio.projectsystem.managed!AbstractSpecialFolderProjectTreePropertiesProvider.CalculatePropertyValues	  9.8	   1,368	 1,368	  0.0	  1
|+ microsoft.visualstudio.projectsystem.managed!AppDesignerFolderProjectTreePropertiesProvider.get_IsSupported           	  9.6	   1,340	 1,340	  0.0	  0
||+ microsoft.visualstudio.projectsystem.managed.vs!ProjectDesignerService.get_SupportsProjectDesigner                   	  9.5	   1,331	 1,331	  0.0	  0

### PR DESCRIPTION
**Pull request by mavasani**
…s appdesigner or not.

This avoids invoking into the hierarchy while calculating property values for app designer folders and should address the CPS regression in #618.
Fixes #618.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/project-system @lifengl for review.

Tagging @MattGertz for ask mode approval.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/ProjectDesignerService.cs [hunk 1]
```diff
@@ -16,19 +16,18 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
...
         private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly bool _supportsProjectDesigner;
 
         [ImportingConstructor]
         public ProjectDesignerService(IUnconfiguredProjectVsServices projectVsServices)
         {
             Requires.NotNull(projectVsServices, nameof(projectVsServices));
 
             _projectVsServices = projectVsServices;
+            _supportsProjectDesigner = _projectVsServices.VsHierarchy.GetProperty(VsHierarchyPropID.SupportsProjectDesigner, defaultValue: false);
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/ProjectDesignerService.cs]: Can this value change within a session? Tagging @lifengl

**Comment by MattGertz** [comment]
(I'm going to wait on this one until feedback is addressed.)

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/ProjectDesignerService.cs (hunk 1 above)
Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/ProjectDesignerService.cs]: From my understanding, this value is computed through [this](http://index/?rightProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&file=PropertyPages%5cVsProjectDesignerPageService.cs&line=50) property, which basically checks if there are any ProjectDesignerPageProviders exported at UnconfiguredProject level, which cannot change during the lifetime of an unconfigured project.

**Comment by MattGertz** [comment]
Approved pending Lifeng's signoff.

**Comment by lifengl** [comment]
looks good to me.  Let me know it is merged into the RC branch.

**Comment by mavasani** [comment]
Thanks Lifeng. @srivatsn Will we be inserting roslyn-project-system into d15prerel today?

**Comment by srivatsn** [comment]
Yes.. can you sync with Nat\Vlad.

**Pull request merged**